File: src/ca/qc/bdeb/inf203/model/combatants/VeggieHitler.java
package ca.qc.bdeb.inf203.model.combatants;

import ca.qc.bdeb.inf203.model.Action;
import ca.qc.bdeb.inf203.model.Combattant;
import ca.qc.bdeb.inf203.model.Entite;
import ca.qc.bdeb.inf203.model.Etat;
import static ca.qc.bdeb.inf203.model.Etat.ATTAQUE;
import static ca.qc.bdeb.inf203.model.Etat.DEPLACEMENT;
import ca.qc.bdeb.inf203.model.RepresentationImage;
import java.awt.Rectangle;
import java.util.HashMap;
import javax.swing.AbstractAction;

/**
 * Veggie
 *
 * @author Guillaume Riou, Nicolas Hurtubise
 */
public class VeggieHitler extends Combattant {

    public VeggieHitler() {
        super();
        initialise();
    }

    @Override
    protected final void initialise() {
        super.initialise();

        this.attaque = 1;
        this.gentil = false;
        this.hitbox.height = 80;
        this.hitbox.width = 60;
        this.animationFrameRate = 5;

        this.lineOfSight = new Rectangle();
        this.lineOfSight = new Rectangle();

        this.vie = 1;

        this.nbrImagesAnimation.put(Etat.DEPLACEMENT, 6);
        this.nbrImagesAnimation.put(Etat.ATTAQUE, 1);

        this.vitesseAction.put(Action.DEPLACEMENT, -70f);
        this.vitesseAction.put(Action.ATTAQUE, 0.6f);
        this.vitesseAction.put(Action.ACTION, 0.7f);

        this.sprites.put(Etat.DEPLACEMENT, new RepresentationImage(new String[]{"veggies", "hitler", "walking"}));
        this.sprites.put(Etat.ATTAQUE, new RepresentationImage(new String[]{"veggies", "hitler", "walking"}));
    }

    @Override
    public Entite tic() {
        // effectuer l'action par rapport à l'état.
        Entite swas = null;
        switch (etat) {
            case DEPLACEMENT:
                deplacer(getNbrActions(Action.DEPLACEMENT));
            case ATTAQUE:
                swas = action(getNbrActions(Action.ACTION));
                break;
        }

        return swas;
    }

    /**
     * 
     *
     * @return une nouvelle Swastika.
     */
    @Override
    public Entite action(int nbrFois) {

        if (nbrFois > 0) {

            Swastika swastika = new Swastika();
            swastika.getHitbox().setLocation(this.hitbox.x + 30, this.hitbox.y + 30);

            swastika.setGentil(this.isGentil());

            HashMap<Action, Float> vitesses = new HashMap<>();

            vitesses.put(Action.DEPLACEMENT, (this.gentil ? 1 : -1) * Math.abs(swastika.getVitesseAction().get(Action.DEPLACEMENT)));
            vitesses.put(Action.ATTAQUE, swastika.getVitesseAction().get(Action.ATTAQUE));

            swastika.setVitesseAction(vitesses);

            return swastika;
        }

        return null;
    }
}

File: assets/js/model/combatants/veggie-hitler.js
/**
 * Veggie Hitler
 *
 * @author Guillaume Riou, Nicolas Hurtubise
 */
function VeggieHitler() {
    Combattant.apply(this);
    this.combattant_init();
    this.init();
}

VeggieHitler.prototype = new Combattant();

VeggieHitler.prototype.init = function() {
    this.attaque = 1;
    this.gentil = false;
    this.hitbox.h = 80;
    this.hitbox.w = 60;
    this.animationFrameRate = 5;
    this.etat = navigator.Etat.DEPLACEMENT;

    this.lineOfSight = new Rectangle();
    this.lineOfSight = new Rectangle();

    this.vie = 1;

    this.nbrImagesAnimation[navigator.Etat.DEPLACEMENT] = 6;
    this.nbrImagesAnimation[navigator.Etat.ATTAQUE] = 1;

    this.vitesseAction[navigator.Action.DEPLACEMENT] = -70;
    this.vitesseAction[navigator.Action.ATTAQUE] = 0.6;
    this.vitesseAction[navigator.Action.ACTION] = 0.7;

    this.sprites[navigator.Etat.DEPLACEMENT] = new RepresentationImage(["veggies", "hitler", "walking"]);
    this.sprites[navigator.Etat.ATTAQUE] = new RepresentationImage(["veggies", "hitler", "walking"]);
};

VeggieHitler.prototype.tic = function() {
    var swas = null;
    switch (this.etat) {
        case navigator.Etat.DEPLACEMENT:
            this.deplacer(this.getNbrActions(navigator.Action.DEPLACEMENT));
        case navigator.Etat.ATTAQUE:
            swas = this.action(this.getNbrActions(navigator.Action.ACTION));
            break;
    }

    return swas;
};

/**
 * @return Entite une nouvelle Swastika.
 */
VeggieHitler.prototype.action = function(nbrFois) {
    if (nbrFois) {
        var swastika = new Swastika();
        swastika.hitbox.x = this.hitbox.x + 30;
        swastika.hitbox.y = this.hitbox.y + 30;

        swastika.gentil = this.gentil;

        var vitesses = {};

        vitesses[navigator.Action.DEPLACEMENT] = (this.gentil ? 1 : -1) * Math.abs(swastika.vitesseAction[navigator.Action.DEPLACEMENT]);
        vitesses[navigator.Action.ATTAQUE] = swastika.vitesseAction[navigator.Action.ATTAQUE];

        swastika.vitesseAction = vitesses;

        return swastika;
    }

    return null;
};
